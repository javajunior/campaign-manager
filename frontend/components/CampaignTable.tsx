import React from 'react'
import { Table } from 'semantic-ui-react'

// The campaign table skeleton
const CampaignTable = props => (
  <Table>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell>ID</Table.HeaderCell>
        <Table.HeaderCell>Campaign Name</Table.HeaderCell>
        <Table.HeaderCell>Start Date</Table.HeaderCell>
        <Table.HeaderCell>End Date</Table.HeaderCell>
        <Table.HeaderCell>Status</Table.HeaderCell>
        <Table.HeaderCell>Budget</Table.HeaderCell>
      </Table.Row>
    </Table.Header>

    <Table.Body>{props.children}</Table.Body>

    <Table.Footer>
      <Table.Row>
        <Table.HeaderCell />
        <Table.HeaderCell>{props.data.length} Total Campaigns</Table.HeaderCell>
        <Table.HeaderCell />
        <Table.HeaderCell />
        <Table.HeaderCell></Table.HeaderCell>
        <Table.HeaderCell></Table.HeaderCell>
      </Table.Row>
    </Table.Footer>
  </Table>
)
export default CampaignTable
