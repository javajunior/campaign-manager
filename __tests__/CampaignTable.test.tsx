import * as React from 'react'
import CampaignTable from '../components/CampaignTable'
import InitialData from '../public/initialData'
import renderer from 'react-test-renderer'

it('renders the "CampaignTable" component correctly', () => {
  const tree = renderer.create(<CampaignTable data={InitialData} />).toJSON()
  expect(tree).toMatchSnapshot()
})
