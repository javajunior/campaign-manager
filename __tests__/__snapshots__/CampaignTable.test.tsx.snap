// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the "CampaignTable" component correctly 1`] = `
<table
  className="ui table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        ID
      </th>
      <th
        className=""
      >
        Campaign Name
      </th>
      <th
        className=""
      >
        Start Date
      </th>
      <th
        className=""
      >
        End Date
      </th>
      <th
        className=""
      >
        Status
      </th>
      <th
        className=""
      >
        Budget
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  />
  <tfoot
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      />
      <th
        className=""
      >
        10
         Total Campaigns
      </th>
      <th
        className=""
      />
      <th
        className=""
      />
      <th
        className=""
      />
      <th
        className=""
      />
    </tr>
  </tfoot>
</table>
`;
